### ISSUE TITLE
Backup System

### ISSUE THREAD

**Issue by shogunpurple** [Budibase/budibase#946]
When using budibase as an admin/dev user, I want to be able to perform a backup of my entire system, so that I can migrate to another host, or run regular backups/snapshots for disaster recovery.

**COS**
- Allow users to backup their entire system (DB/MinIO assets) to a pouchDB dump `budi backup --export tarball.tar.gz`
- Allow users to restore an entire system (DB/MinIO assets) from a pouchDB dump `budi backup --import tarball.tar.gz`
- Default to the couchDB URL in `.env` but allow it to be overwritten with the `--url` flag in the `budi backup` command.

When importing, it is an operation that will overwrite existing data that has the same appIds, for example. We must warn the user about this and ensure that they know what this operation does.

**Comment by mike12345567** [comment]
Issue still in discussion, a feature that backs up Couch data to MinIO/S3 would be useful for a generic method of backup that can be utilised across all installation types.

**Comment by theobouwman** [comment]
What is the best way to do a full backup of the system when using docker?

**Comment by jackgray** [comment]
Arrived here while researching methods of version control after realizing there is currently nothing built in to budibase, which I am hoping will be a good fit for my clinical research team. 

I want to use a document-based db. Arrango, ideally but with this new requirement/limitation, it seams the maturity of Mongo might be my best bet.

So far the leads I have found are:
  - Frequent snapshot backups to MinIO. I intended to use MinIO regardless, and it seems to integrate with all the dbs offered by Budibase. I will have to find out if it's a good idea to use the same kubernetes service that budibase launches, or if I need my own pod.

  - MongoDB Document Versioning Pattern (https://www.mongodb.com/blog/post/building-with-patterns-the-document-versioning-pattern) 
            - tracks revisions by adding a revision field in the document and creating a new document on field changes, and a n+1 to the revision number for the changed item.

            - has limitations: cannot track a large number of documents. Cannot track a large amount of changes.

  - MongoDB Change Streams + Kafka Connector, dump the streams into MinIO and figure out a way to access them from budibase.

  - Create budibase automations that duplicate any table with a timestamp when changes are made to it
